hls4ml/templates/quartus/firmware/nnet_utils/nnet_batchnorm_stream.h

```diff
@@ -1,33 +1,100 @@
-//
-//    rfnoc-hls-neuralnet: Vivado HLS code for neural-net building blocks
-//
-//    Copyright (C) 2017 EJ Kreinar
-//
-//    This program is free software: you can redistribute it and/or modify
-//    it under the terms of the GNU General Public License as published by
-//    the Free Software Foundation, either version 3 of the License, or
-//    (at your option) any later version.
-//
-//    This program is distributed in the hope that it will be useful,
-//    but WITHOUT ANY WARRANTY; without even the implied warranty of
-//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-//    GNU General Public License for more details.
-//
-//    You should have received a copy of the GNU General Public License
-//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-//
-
-/*
-*   PLACEHOLDER - The common pass bn_quant.py includes both parallel and streaming BN; streaming is currently not supported in Quartus
-*/
-
 #ifndef NNET_BATCHNORM_STREAM_H_
 #define NNET_BATCHNORM_STREAM_H_
 
 #include "nnet_common.h"
 #include "nnet_helpers.h"
 #include "nnet_mult.h"
+#include "nnet_types.h"
+
+namespace nnet {
+
+// ****************************************************
+//       Streaming Batch Normalization
+// ****************************************************
+template<class data_T, class res_T, typename CONFIG_T>
+void normalize(
+    stream<data_T> &data, 
+    stream<res_T> &res,
+    const typename CONFIG_T::scale_t scale[CONFIG_T::n_in],
```

Comment:
Note that on the Vivado side these are smaller:  CONFIG_T::n_scale_bias. Basically, in n_filt != 1, then a smaller scale and bias is actually used. That change can come later, though, and doesn't need to hold this PR.

Reply:
Okay, should we open an issue, just so we don't lose track of it?

Reply:
#497 already exists and wasn't closed. It was fixed for Vivado. Or is it better to open a new issue?

Reply:
(It is a fix I can make after we merge.)